File: java_pro/src/main/java/jv/com/tencent/cain/design_pattern/bridge/HandsetSoft.java
package jv.com.tencent.cain.design_pattern.bridge;

/**
 * @author : jiangyu
 * @date : 2020/7/11
 * @desc : 桥接设计模式-剥离出来的一个分支
 */
abstract public class HandsetSoft {
    protected String softName;

    public HandsetSoft(String name) {
        this.softName = name;
    }

    public void setSoftName(String softName) {
        this.softName = softName;
    }

    abstract void run();
}

File: java_pro/src/main/kotlin/kt/com/tencent/cain/desigin_pattern/bridge/HandsetSoft.kt
package kt.com.tencent.cain.desigin_pattern.bridge

/**
 * @author : jiangyu
 * @date   : 2020/7/11
 * @desc   : 桥接设计模式-剥离出来的一个分支
 */
abstract class HandsetSoft(var softName: String = "") {
    abstract fun run()
}